src/main/java/com/facebook/openwifirrm/ucentral/UCentralClient.java

```diff
@@ -614,6 +624,39 @@ public void setServiceEndpoint(String service, ServiceEvent event) {
 		}
 	}
 
+	/**
+	 * Validate the token is still valid.
+	 */
+	public boolean validateToken(String token) {
+		if (!tokenCache.containsKey(token)) {
+			HttpResponse<String> response =
+					httpGet("validateToken", OWSEC_SERVICE,
+							Collections.singletonMap("token", token));
+
+			if (!response.isSuccess()) {
+				logger.error("Error: {}", response.getBody());
+				return false;
+			}
+
+			TokenValidationResult result;
+			try {
+				result = gson.fromJson(response.getBody(), TokenValidationResult.class);
+			} catch (JsonSyntaxException e) {
+				String errMsg = String.format(
+						"Failed to deserialize to TokenValidationResult: %s", response.getBody()
+				);
+				logger.error(errMsg, e);
+				return false;
+			}
+
+			tokenCache.put(token, result.tokenInfo.created + result.tokenInfo.expires_in);
+		}
+
+		long expiry = tokenCache.get(token);
```

Comment:
this will be covered by the above comment, but note that this might crash in async/concurrent edge case if the "token" no longer exists (ex. got removed from cache somewhere else) and "null" can't auto-unbox into "long" (will raise NullPointerException).